dask/dataframe/shuffle.py

```diff
@@ -173,14 +180,18 @@ def set_partition(
     shuffle
     partd
     """
+    if isinstance(divisions, list) or isinstance(divisions, tuple):
+        divisions = pd.Series(divisions)
+    if meta is None:
+        meta = pd.Series([0])
     if np.isscalar(index):
         partitions = df[index].map_partitions(
-            set_partitions_pre, divisions=divisions, meta=pd.Series([0])
+            set_partitions_pre, divisions=divisions, meta=meta
         )
```

Comment:
I agree that we shouldn't hardcode this to `pd.Series`.  I'm curious if we could get the `_meta` type from the dataframe itself?  Maybe something like `df[index]._meta`?